### PR TITLE
Shows pipeline owner

### PR DESCRIPTION
**Pull request by raulb**
## Checklist

- [x] Show pipeline owner when it's present
- [x] Update tests
- [x] Update yarn.lock
- [x] Update [CHANGELOG.md](https://github.com/heroku/heroku-pipelines/blob/master/CHANGELOG.md)

## Why

While working on Pipelines Ownership, it's useful to show the pipeline owner, and with the use of `--with-owners`, show the owner of all pipeline couplings.

## What are the acceptance criteria for the change?

- [ ] [DevEx](https://github.com/heroku/devex) gives a :+1:

## How can the change be tested

1. `heroku plugins:link .`
2. Run `pipelines:info` command with a pipeline you own.

## Demonstration

**Before**

```
› heroku pipelines:info sausages
=== sausages
review:     sausages-staging-pr-28
            sausages-staging-pr-30
            sausages-staging-pr-16
            sausages-staging-pr-25
            sausages-staging-pr-26
            sausages-staging-pr-19
            sausages-staging-pr-23
staging:    andouille
            sausages-are-not-paranoid
            sausages-staging
production: sausages-eu
            sausages
            sausages-tokyo
```

**Now**

```
› heroku pipelines:info sausages
name:  sausages
owner: heroku-devex (team)

app name                     stage
───────────────────────────  ──────────
⬢ sausages-staging-pr-16     review
⬢ sausages-staging-pr-19     review
⬢ sausages-staging-pr-23     review
⬢ sausages-staging-pr-25     review
⬢ sausages-staging-pr-26     review
⬢ sausages-staging-pr-28     review
⬢ sausages-staging-pr-30     review
⬢ andouille                  staging
⬢ sausages-are-not-paranoid  staging
⬢ sausages-staging           staging
⬢ sausages                   production
⬢ sausages-eu                production
⬢ sausages-tokyo             production

› heroku pipelines:info sausages --with-owners
name:  sausages
owner: heroku-devex (team)

app name                     stage       owner
───────────────────────────  ──────────  ───────────────────
⬢ sausages-staging-pr-16     review      heroku-devex (team)
⬢ sausages-staging-pr-19     review      heroku-devex (team)
⬢ sausages-staging-pr-23     review      heroku-devex (team)
⬢ sausages-staging-pr-25     review      heroku-devex (team)
⬢ sausages-staging-pr-26     review      heroku-devex (team)
⬢ sausages-staging-pr-28     review      heroku-devex (team)
⬢ sausages-staging-pr-30     review      heroku-devex (team)
⬢ andouille                  staging     joshwlewis (team)
⬢ sausages-are-not-paranoid  staging     heroku-devex (team)
⬢ sausages-staging           staging     heroku-devex (team)
⬢ sausages                   production  heroku-devex (team)
⬢ sausages-eu                production  heroku-devex (team)
⬢ sausages-tokyo             production  heroku-devex (team)
```

Easier to parse:

```
› heroku pipelines:info sausages | grep review
sausages-staging-pr-16     review
sausages-staging-pr-19     review
sausages-staging-pr-23     review
sausages-staging-pr-25     review
sausages-staging-pr-26     review
sausages-staging-pr-28     review
sausages-staging-pr-30     review
```

## Who's Affected

All users.

## Blockers & upstream dependencies

None.

### CODE REVIEW

File: commands/pipelines/info.js [hunk 1]
```diff
@@ -22,32 +32,49 @@ module.exports = {
...
+    {name: 'with-owners', description: 'shows owner of every app', hidden: true}
   ],
   run: cli.command(co.wrap(function* (context, heroku) {
     const pipeline = yield disambiguate(heroku, context.args.pipeline)
 
-    const apps = yield listPipelineApps(heroku, pipeline.id)
+    let apps = yield listPipelineApps(heroku, pipeline.id)
+    const reviewApps = sortBy(apps.filter(app => app.coupling.stage === 'review'), ['name'])
+    const stagingApps = sortBy(apps.filter(app => app.coupling.stage === 'staging'), ['name'])
+    const productionApps = sortBy(apps.filter(app => app.coupling.stage === 'production'), ['name'])
```
Comment (joshwlewis) [inline commands/pipelines/info.js]: `developmentApps` is missing here. I ran into this today by chance, debugging customer issues and couldn't figure out why their review app parent wasn't in the pipeline. It was in the development lane...

File: commands/pipelines/info.js [hunk 2]
```diff
@@ -3,8 +3,18 @@
...
 const cli = require('heroku-cli-util')
 const disambiguate = require('../../lib/disambiguate')
-const stageNames = require('../../lib/stages').names
 const listPipelineApps = require('../../lib/api').listPipelineApps
+const getTeam = require('../../lib/api').getTeam
+const sortBy = require('lodash.sortby')
+
+// For user pipelines we need to use their couplings to determine user email
+function getUserPipelineOwner (apps, userId) {
+  for (let app in apps) {
```
Comment (joshwlewis) [inline commands/pipelines/info.js]: It's not likely, but possible that the none of the apps the current user is collaborated on are owned by the pipeline owner. In other words, there may not be an app in the list with `userId` as the owner here. I'd suggest returning the `userId` here if we can't resolve an email.

File: commands/pipelines/info.js (hunk 1 above)
Comment (raulb) [reply commands/pipelines/info.js]: @joshwlewis oh!, thought we wanted to hide them on purpose. Will add

File: commands/pipelines/info.js (hunk 2 above)
Comment (raulb) [reply commands/pipelines/info.js]: I think what you mean is that if pipeline owner doesn't own any application in the pipeline, but I agree, we should at least return the uuid there.

Comment (raulb) [reply commands/pipelines/info.js]: @joshwlewis https://github.com/heroku/heroku-pipelines/pull/67/commits/09ccd4070516eee9b76f3918b43548ca4abca8c8

**Pull request merged**